src/main.cpp

```diff
@@ -166,6 +192,36 @@ void loop() {
 #else
   bleServer->loop();
 #endif
+
+#ifdef WIFI_ENABLED
+  ArduinoOTA.handle();
+
+  WiFiClient client = WifiServer.available();
+  
+  if (client) {
+    if(client.connected())
+    {
+      //Serial.println("Client Connected");
+      Logger::error(LOG_TAG_BLESERVER, "Client Connected");
+    }
+    
+    while(client.connected()){      
+      while(client.available()>0){
+        // read data from the connected client
+        vesc.write(client.read()); 
+      }
+      //Send Data to connected client
+      while(vesc.available()>0)
+      {
+        client.write(vesc.read());
+      }
+    }
+    client.stop();
+    //Serial.println("Client disconnected");    
+    Logger::error(LOG_TAG_BLESERVER, "Client disconnected");
+  }
```

Comment:
**Suggestion:** The WiFi client handling code in the main loop could be refactored into a separate function for better readability and maintainability.
```suggestion
#ifdef WIFI_ENABLED
  handleWifiClient();
#endif // WIFI_ENABLED

void handleWifiClient() {
  ArduinoOTA.handle();
  WiFiClient client = WifiServer.available();
  if (client) {
    if(client.connected()) {
      Logger::error(LOG_TAG_BLESERVER, "Client Connected");
    }
    while(client.connected()){      
      while(client.available()>0){
        vesc.write(client.read()); 
      }
      while(vesc.available()>0) {
        client.write(vesc.read());
      }
    }
    client.stop();
    Logger::error(LOG_TAG_BLESERVER, "Client disconnected");
  }
}
```